jax/experimental/pallas/ops/attention.py

```diff
@@ -91,6 +92,75 @@ def body(start_k, carry):
   acc = acc.astype(o_ref.dtype)
   pl.store(o_ref, (pl.dslice(start_q * block_q, block_q), pl.dslice(None)), acc)
 
+def _mha_forward_kernel(
```

Comment:
For reference, this is the "improved" softmax computation proposed in flash attention 2. However, we found it to be slower on a RTX 4070 Laptop GPU

Reply:
We can remove it before merge.

Reply:
Moved into a config variable.